portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,6 +60,25 @@ private String getParameter(HttpServletRequest request, String name, String defa
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment_null = request.getParameter("limit");
+    
+    Query query = new Query("Comments");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> messages = new ArrayList<String>();
+
+    for (Entity entity : results.asIterable()) {
+      String message = (String) entity.getProperty("message");
+        if(comment_null != null){
+         int comment_num = (Integer.valueOf(request.getParameter("limit")));
+          if(messages.size() == comment_num){
+            break;
+          }
+        }
```

Comment:
This block is indented too far I think. It should line up right under the "String message"